test/parser/ultimate_guitar_chordsheet.txt

```diff
@@ -0,0 +1,45 @@
+[Verse 1]
```

Comment:
Just thinking "out loud": it seems to me an Ultimate Guitar sheet is _almost_ identical to a regular chord sheet, except for the fact that a section indicator like "Chorus" or "Verse" is wrapped in brackets. Might there be a way to share more code with `ChordSheetParser`? It seems much code for such a small difference, but you should correct me if I'm wrong about that. What are your thoughts about that?

Reply:
I agree, it is a lot of additional code.
Plus, the rather wobbly subclassing making the ultimate parser rely heavily on the internals of the chordsheet parser.

I didn't want to touch the existing parser at first, which is why I opted for new file.

Here are some options I see:
1. Subclassing (as is done right now in the PR)
  a. Maybe we could add the functionality to the chord sheet parser. If the ultimate tags in `[]` brackets are there, great. If not, everything stays as it is.
2. One big ChordSheetParser that can handle ultimate `[] ` tags if they occur. Add the ultimate guitar section code to the `ChordSheetParser` class.
3. Use a better architectural pattern than inheritance/subclassing to add the ultimate guitar section behaviour.
  a. E.g. strategy pattern. Pass a section detector strategy as a constructor param to the chord sheet parser (default: `NoSectionDetector`). The detector gets to peek at each line before the main parser handles it and can choose to start/end the section.
  b. For ease of use, this behaviour might still be exposed as is.
```js
class UltimateGuitarParser {
  constructor(options) {
    this.delegate = new ChordSheetParser(options, new UltimateSectionDetector());
  }
  
  parse() {
    return this.delegate.parse();
  }
}
```

Reply:
I think delegation could be a good pattern, but maybe the other way around? The UltimateGuitar parser can take the lead, but it could hand an entire section over to `ChordSheetParser` to get it parsed. This way you can reuse functionality via the public API of `ChordSheetParser`. What do you think?